### DISCUSSION TITLE
inside snapcraft.yaml, vcpkg, gcc-9

### DISCUSSION THREAD

**Discussion by zweistein-frm2** [microsoft/vcpkg#19751] [Q&A]
Hello,

I am trying to use vcpkg inside a snapcraft.yaml file. 
vcpkg is used by my c++ project based on gcc-9.

It seems that inside snapcraft.yaml all references are lost and I have to give the directories by hand.
The toolchain file does not detect the libraries present.
See example:

cmake \
        -DCMAKE_INSTALL_PREFIX=$SNAPCRAFT_PART_INSTALL/usr/local \
        -DCMAKE_BUILD_TYPE=Release \
        -DVCPKG_TARGET_TRIPLET=x64-linux \
        -D_VCPKG_ROOT_DIR=/root/parts/vcpkg/build \
        -DCMAKE_TOOLCHAIN_FILE=/root/parts/vcpkg/build/scripts/buildsystems/vcpkg.cmake \
        -DOpenCV_DIR=/root/parts/vcpkg/build/installed/x64-linux/share/opencv \

Best regards
Andreas

Full file:  snapcraft.yaml (source is at github.com/zweistein-frm2/CHARMing/charm

[snapcraft.yaml.txt](https://github.com/microsoft/vcpkg/files/4629088/snapcraft.yaml.txt)

**Comment by PhoebeHui** [comment]
@zweistein-frm2, do you solve the problem? what's failures it produce?

BTW, I don't think you need to pass the following args to cmake in your yaml file

        -DOpenCV_DIR=/root/parts/vcpkg/build/installed/x64-linux/share/opencv \
        -DProtobuf_DIR=/root/parts/vcpkg/build/installed/x64-linux/share/protobuf \
        -DBoost_INCLUDE_DIR=/root/prime/include  \
        -DProtobuf_INCLUDE_DIRS=/root/parts/vcpkg/build/installed/x64-linux/include  \
        -Dunofficial-libuv_DIR=/root/parts/vcpkg/build/installed/x64-linux/share/unofficial-libuv \
        -Dmagic_enum_DIR=/root/parts/vcpkg/build/installed/x64-linux/share/magic_enum \
